chia/data_layer/data_store.py

```diff
@@ -655,3 +734,55 @@ async def get_proof_of_inclusion_by_key(
         async with self.db_wrapper.locked_transaction(lock=lock):
             node = await self.get_node_by_key(key=key, tree_id=tree_id, lock=False)
             return await self.get_proof_of_inclusion_by_hash(node_hash=node.hash, tree_id=tree_id, lock=False)
+
+    async def answer_server_query(
```

Comment:
I have a feeling there's a better name for this function. What does it do exactly?
Does it return the subtree rooted in `node_hash` in the tree `tree_id` (is that the same as launcher ID?)
But why does it need the `root_hash`? The subtree is *defined* by the `node_hash`, it doesn't matter what the root is. But maybe I misunderstand what this function does.

Reply:
this function returns left-to-right ordering of the tree. when we're the rightmost path in `node_hash` subtree, but still need to answer more nodes, we need to go up in the ancestors and then right, to find the next subtree.

Reply:
so there is probably a better name than "answer_server_query()" then. So, do we return the subtree rooted in `node_hash` but then uncles from `node_hash` to the `root`, to prove `node_hash` is a part of the tree?
The term "ancestors" I thought meant previous states of the tree, is that what it refers to? or does it refer to parent and uncle nodes in the tree?

Reply:
yes, it's parents and uncles. I'll rename it `get_left_to_right` ordering. no, the intend is not to prove node_hash is part of root. it's to get more nodes after we've finished the subtree. this saves queries: say we queried a subtree with 3 nodes, but if server wants 5000 answers, it'll go up and right in the tree to find other 4997 nodes.